components/entities/Dragon.js

```diff
@@ -0,0 +1,42 @@
+import React from "react";
+import { ImageBackground } from "react-native";
+import Matter from "matter-js";
+import { useEntity } from "../../hooks/useEntity";
+
+
+const Dragon = ({ body }) => {
+  const { getDimmensions } = useEntity();
+
+  const { widthBody, heightBody, xBody, yBody } = getDimmensions(body);
```

Comment:
getDimmensiosn takes in body and returns the same variables in each of your entities, so using this hook will consolidate your logic into one page.